File: integration-tests/TestAPI-bindings/kotlin/src/test/kotlin/com/cricut/testapi/StringTests.kt
package com.cricut.testapi

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class StringTests {
    @Test
    fun testStringValues() {
        assertEquals(Strings.simple, "Hello")
        assertEquals(Strings.accent, "Olá")
        assertEquals(Strings.script, "こんにちは")
        assertEquals(Strings.chinese, "你好")
        assertEquals(Strings.chineseBMP, "豈更車賈滑")
        assertEquals(Strings.chineseSIP, "\uD840\uDC00\uD840\uDC02\uD840\uDC03\uD840\uDC04")
        assertEquals(Strings.emoji, "🤯🐶🍓")
        assertEquals(Strings.emojiMulti, "👨‍👩‍👧‍👦👍🏿🇺🇸")
    }

    @Test
    fun testStringEcho() {
        assertEquals(Strings.echo(Strings.simple), Strings.simple)
        assertEquals(Strings.echo(Strings.accent), Strings.accent)
        assertEquals(Strings.echo(Strings.chinese), Strings.chinese)
        assertEquals(Strings.echo(Strings.chineseBMP), Strings.chineseBMP)
        assertEquals(Strings.echo(Strings.chineseSIP), Strings.chineseSIP)
        assertEquals(Strings.echo(Strings.emoji), Strings.emoji)
        assertEquals(Strings.echo(Strings.emojiMulti), Strings.emojiMulti)
    }
}

File: integration-tests/TestAPI-bindings/c-sharp/TestAPI.Tests/StringTests.cs
using System;
using Xunit;
using Cricut.TestAPI;

namespace TestAPI.Tests {
    public class StringTests {
        [Fact]
        void testStringValues() {
            Assert.Equal("Hello", Strings.simple);
            Assert.Equal("Olá", Strings.accent);
            Assert.Equal("こんにちは", Strings.script);
            Assert.Equal("你好", Strings.chinese);
            Assert.Equal("豈更車賈滑", Strings.chineseBMP);
            Assert.Equal("\uD840\uDC01\uD840\uDC02\uD840\uDC03\uD840\uDC04", Strings.chineseSIP);
            Assert.Equal("🤯🐶🍓", Strings.emoji);
            Assert.Equal("👨‍👩‍👧‍👦👍🏿🇺🇸", Strings.emojiMulti);
        }

        [Fact]
        void testStringEcho() {
            Assert.Equal(Strings.simple, Strings.echo(Strings.simple));
            Assert.Equal(Strings.accent, Strings.echo(Strings.accent));
            Assert.Equal(Strings.chinese, Strings.echo(Strings.chinese));
            Assert.Equal(Strings.chineseBMP, Strings.echo(Strings.chineseBMP));
            Assert.Equal(Strings.chineseSIP, Strings.echo(Strings.chineseSIP));
            Assert.Equal(Strings.emoji, Strings.echo(Strings.emoji));
            Assert.Equal(Strings.emojiMulti, Strings.echo(Strings.emojiMulti));
        }
    }
}
